docs/tutorials/embassy/index.md

```diff
@@ -0,0 +1,148 @@
+
+> Note: You may be prompted to install [Visual Studio C++ Build tools](https://visualstudio.microsoft.com/visual-cpp-build-tools/). If so, follow the instructions from the previous link.
+
+The last step is to run `rustup --version` in terminal. If everything went well, you should see an output similar to this:
+
+```shell
+rustup 1.26.0 (5af9b9484 2023-04-05)
+info: This is the version for the rustup toolchain manager, not the rustc compiler.
+info: The currently active `rustc` version is `rustc 1.76.0 (07dca489a 2024-02-04)`
+```
+
+### `elf2uf2-rs`
+
+This tool is needed to be able to program the board over `USB`. In order to install it, run the following in your terminal:
+
+```shell
+cargo install elf2uf2-rs
+```
+
+Then, run `elf2uf2-rs --help`. If it was correctly installed, you should see something similar to this in your terminal:
+
+```shell
+Usage: elf2uf2-rs [OPTIONS] <INPUT> [OUTPUT]
+
+Arguments:
+  <INPUT>   Input file
+  [OUTPUT]  Output file
+
+Options:
+  -v, --verbose  Verbose
+  -d, --deploy   Deploy to any connected pico
+  -s, --serial   Connect to serial after deploy
+  -h, --help     Print help
+```
+
+### `probe-rs`
+
+This tool is an embedded debugging and target interaction toolkit. It enables its user to program and debug microcontrollers via a debug probe.
+
+```shell
+cargo install probe-rs --features cli --locked
+```
+
+If you are on **Linux** you will also need to add this [`udev`](https://probe.rs/files/69-probe-rs.rules) file in `/etc/udev/rules.d`. Then, run:
+
+```shell
+udevadm control --reload # to ensure the new rules are used.
+
+udevadm trigger # to ensure the new rules are applied to already added devices.
+```
+
+### VSCode Extension (Optional)
+
+For a better experience, go ahead and install the **Debugger for probe-rs** extension in the Microsoft Extension Marketplace. This will make debugging the program running on the `RP2040` as easy as debugging a Rust program running on your host machine.
+
+## Flashing over USB
+
+### Compiling
+
+You will need to compile your executable specifically for the `RP2040` chip. This chip is based on the **ARM Cortex M0+** architecture, so we will need to specify our target when compiling. We can do that in multiple ways:
+
+* using a `.cargo/config.toml` file:
+
+```toml
+[build]
+target = "thumbv6m-none-eabi"
+```
+
+* passing it as a parameter to Cargo:
+
+```shell
+cargo build --release --target thumbv6m-none-eabi
+```
+
+### Flashing
+
+To flash a program to the Pi Pico via USB, it needs to be in `USB mass storage device mode`. To put it in this mode, you need to **hold the `BOOTSEL` button down**  while connecting it to your PC. Connecting and disconnecting the USB can lead to the port getting damaged, so we conveniently attached a reset button on the breadboard included on the **Pico Explorer Base**. Now, to make it reflashable again, just press the two buttons simultaneously.
```

Comment:
```suggestion
To flash a program to the Raspberry Pi Pico via USB, it needs to be in *USB mass storage device mode*. To put it in this mode, you need to **hold the `BOOTSEL` button down**  while connecting it to your PC. Connecting and disconnecting the USB can lead to the port getting damaged, so we conveniently attached a reset button on the breadboard included on the **Pico Explorer Base**. Now, to make it reflashable again, just press the two buttons simultaneously.
```

Reply:
Want the USB mass storage device mode in *italics* or **bolded**?